src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -611,21 +612,29 @@ def _get_single_device_connection_string(client, hub_name, device_id, resource_g
     device = device_client.get(device_id)
     if device is None:
         raise CLIError("Device {} not found.".format(device_id))
-
-    conn_str_template = "HostName={0}.azure-devices.net;DeviceId={1};{2}={3}"
+    server_suffix = _get_device_dns_suffix()
+    conn_str_template = "HostName={0}{1};DeviceId={2};{3}={4}"
     keys = device.authentication.symmetric_key
     if any([keys.primary_key, keys.secondary_key]):
         key = keys.secondary_key if key_type == KeyType.secondary else keys.primary_key
         if key is None:
             raise CLIError("{0} key not found.".format(key_type))
-        return conn_str_template.format(hub_name, device_id, 'SharedAccessKey', key)
+        return conn_str_template.format(hub_name, server_suffix, device_id, 'SharedAccessKey', key)
     else:
-        return conn_str_template.format(hub_name, device_id, 'x509', 'true')
+        return conn_str_template.format(hub_name, server_suffix, device_id, 'x509', 'true')
+
+
+def _get_device_dns_suffix():
+    # Allow dns suffix to be overridden by environment variable for testing purposes
+    from os import getenv
+    from azure.cli.core._profile import get_active_cloud
+    return getenv('_AZURE_CLI_IOT_DNS_SUFFIX', default=get_active_cloud().suffixes.iot_device_hostname)
```

Comment:
`get_active_cloud` takes  a recommended argument of `cli_ctx`.
You can get this by passing `cli_ctx` from the command handler (e.g. `iot_hub_sku_list`) by adding a first argument of `cmd`.

e.g. for `iot_hub_sku_list `:
```
def iot_hub_sku_list(cmd, client, ...):
    server_suffix = _get_device_dns_suffix(cmd.cli_ctx)

def _get_device_dns_suffix(cli_ctx):
    from os import getenv
    from azure.cli.core._profile import get_active_cloud
    return getenv('_AZURE_CLI_IOT_DNS_SUFFIX', default=get_active_cloud(cli_ctx).suffixes.iot_device_hostname)
```